api/src/main/java/org/apache/iceberg/catalog/Catalog.java

```diff
@@ -58,9 +60,50 @@ Table createTable(
       TableIdentifier identifier,
       Schema schema,
       PartitionSpec spec,
+      SortOrder sortOrder,
       String location,
       Map<String, String> properties);
 
+  /**
+   * Create a table.
+   *
+   * @param identifier a table identifier
+   * @param schema a schema
+   * @param spec a partition spec
+   * @param location a location for the table; leave null if unspecified
+   * @param properties a string map of table properties
+   * @return a Table instance
+   * @throws AlreadyExistsException if the table already exists
+   */
+  default Table createTable(
```

Comment:
This seems to get out of hand. We may want to switch to a builder here.

Reply:
I think that would definitely make it a little cleaner here and in all of the related "create" methods

Reply:
@rdsr @shardulm94 @rdblue what do you guys think?

Reply:
I'm +1. But this is a public API, if we move to a builder we'd have to keep the current methods around with deprecated annotation.

Reply:
Yeah, I think we can map existing methods into the builder and keep the compatibility.

Reply:
We only double the number of implementations every time we add a new config option...

+1 to a builder, I think that would be helpful.

Reply:
Submitted PR#1409 for this.